lightning_client.go

```diff
@@ -2703,3 +2830,75 @@ func (s *lightningClient) NetworkInfo(ctx context.Context) (*NetworkInfo,
 		NumZombieChans:       resp.NumZombieChans,
 	}, nil
 }
+
+// AcceptorFunction is the signature used for functions passed to our channel
+// acceptor.
+type AcceptorFunction func(*AcceptorRequest) (*AcceptorResponse, error)
+
+// AcceptChannel create a channel acceptor using the accept function
+// passed in.
+func (s *lightningClient) AcceptChannel(ctx context.Context,
+	accept AcceptorFunction) (chan error, error) {
+
+	acceptStream, err := s.client.ChannelAcceptor(
+		s.adminMac.WithMacaroonAuth(ctx),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	errChan := make(chan error, 1)
+
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+
+		for {
+			request, err := acceptStream.Recv()
```

Comment:
Should we select on `ctx.Done()` before this, just in case we get that directly after sending a message. Then we can avoid getting an error on the `Recv()` and quit nicely.

Reply:
sgtm!